UT4MasterServer/Controllers/SessionController.cs

```diff
@@ -37,118 +38,112 @@ public class SessionController : JsonAPIController
 	[AuthorizeBasic]
 	[HttpPost("token")]
 	public async Task<IActionResult> Authenticate(
-		[FromForm(Name = "grant_type")] string grantType,
-		[FromForm(Name = "includePerms")] bool? includePerms,
-		[FromForm(Name = "code")] string? code,
-		[FromForm(Name = "exchange_code")] string? exchangeCode,
-		[FromForm(Name = "refresh_token")] string? refreshToken,
-		[FromForm(Name = "username")] string? username,
-		[FromForm(Name = "password")] string? password)
+		[FromBody] AuthenticateRequest request)
 	{
 		if (User.Identity is not EpicClientIdentity user)
 			return Unauthorized();
 
 		EpicID clientID = user.Client.ID;
 		Session? session = null;
 		Account? account = null;
-		switch (grantType)
+		switch (request.GrantType)
 		{
 			case "authorization_code":
-			{
-				if (code != null)
-				{
-					var codeAuth = await codeService.TakeCodeAsync(CodeKind.Authorization, code);
-					if (codeAuth != null)
-						session = await sessionService.CreateSessionAsync(codeAuth.AccountID, clientID, SessionCreationMethod.AuthorizationCode);
-				}
-				else
 				{
-					return ErrorInvalidRequest("code");
+					if (request.Code != null)
+					{
+						var codeAuth = await codeService.TakeCodeAsync(CodeKind.Authorization, request.Code);
```

Comment:
Fix indentation and curly brackets

Reply:
I autoformatted it, if you want code styles enforced include them in the project